### ISSUE TITLE
Import eases individually in GSAP3 + Typescript

### ISSUE THREAD

**Issue by Lange** [greensock/GSAP#321]
_(Originally posted at https://greensock.com/forums/topic/21892-gsap-3-import-eases-same-as-gsap-2/)_

Hi there,

I'm upgrading from GSAP 2 to GSAP 3 in a Typescript codebase. To retain strong typing and autocomplete, I actually liked having to import each ease object explicitly, as follows:

```ts
import {TweenLite, Power1} from 'gsap'
```

I would like to do the same in GSAP3. It [appears that the actual library does export the ease objects](https://github.com/greensock/GSAP/blob/master/src/index.js#L14-L31), such that doing this would be supported:

```ts
import gsap, {Power1} from 'gsap'
```

However, the included typedef does not list these ease objects as part of the export. Is there a particular reason for this? Thanks!

**Comment by dipscom** [comment]
Hi Lange,

The library does contain a bunch of ease objects, the only reason they have not made it to the definitions yet is that we ran out of time to add them into the file before the release.

I should be able to look into adding them tonight, you can then test them out and, if you're happy with them, they would be included in the next minor update.

**Comment by Lange** [comment]
@dipscom Sounds great! Just ping me, I'll be around.

**Comment by dipscom** [comment]
@Lange here's a quick preivew with (what I think is) the exposed eases.

https://www.dropbox.com/s/t3uynf73hwamr0x/gsap.d.ts?dl=0

I'm no expert in Typescript so, any educating and tips is very welcome.

**Comment by Lange** [comment]
Thanks @dipscom!

Here's the issues I've noticed so far:

1. There's no general `Ease` type exported or available from `GSAPStatic`. This type is used for things like this:

    ```ts
    let dynamicEase: Ease;
    if (foo) {
        dynamicEase = Power1.easeOut;
    else {
        dynamicEase = Circ.easeIn;
    }
    ```

2. The types aren't quite right. Currently, each ease is typed as `() => number`. This is incorrect, and the type would instead be something like this: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/gsap/Ease.d.ts

**Comment by OSUblake** [comment]
@Lange 

We are working on the definitions. Check back in a week or two. 

If you need something that isn't provided at the moment, you are going to to have to provide your own solution. Here's how the easing will look.

```ts
type EaseFunction = (progress: number) => number;
  
interface Ease {
  easeIn: EaseFunction;
  easeOut: EaseFunction;
  easeInOut: EaseFunction;
}

interface ElasticEase {
  (amplitude?: number, period?: number): EaseFunction;
  config(amplitude?: number, period?: number): EaseFunction;
}

declare const Power1: Ease;
declare const Power2: Ease;
declare const Elastic: {
  easeIn: ElasticEase;
  easeOut: ElasticEase;
  easeInOut: ElasticEase;
}
...
```

**Comment by Lange** [comment]
@OSUblake Totally fine, I can stick on GSAP 2 for now. I'm not in any particular rush. Thanks for y'alls hard work on making first-party typedefs! I greatly appreciate it ❤

**Comment by OSUblake** [comment]
Just create your typings file, and add those easings so you can continue using v3.

**Comment by dipscom** [comment]
@Lange much to learn, I see.

I'll be taking the template Blake has shown and your links and have another go. We'll get there in the end.

**Comment by jackdoyle** [comment]
Should be resolved in the latest release (3.0.2)
